submissions/5mountains/tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,105 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Resident {
+  constructor(species, name, gender, saying, friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+    this.friends = friends;
+  }
+  introducing() {
+    return `Wonderful creature - <b>${
+      this.species
+    }</b>, whose name is <b>${
+      this.name
+    }</b>! Usually the greeting is <b>${
+      this.saying
+    }</b></i> and friends of this inhabitant are <b>${
+      this.friends.join(', ')
+    }</b>. By the wat, this member is owner of`;
+  }
+}
+
+class Human extends Resident {
+  constructor(name, gender, saying, friends) {
+    super('human', name, gender, saying, friends);
+    this.legs = 2;
+    this.hands = 2;
+  }
+  introducing() {
+    return (
+      super.introducing() +
+      ` <b>${this.legs} legs</b> and <b>${this.hands} hands</b>!`
+    );
+  }
+}
+
+class Animal extends Resident {
+  constructor(species, name, gender, saying, friends) {
+    super(species, name, gender, saying, friends);
+    this.paws = 4;
+  }
+  introducing() {
+    return super.introducing() + ` <b>${this.paws} paws</b>!`;
+  }
+}
+
+class Dog extends Animal {
+  constructor(name, gender) {
+    super('dog', name, gender, 'woof-woof!');
+    this.friends = ['Tom', 'Jenny'];
+  }
+}
+
+class Cat extends Animal {
+  constructor(name, gender) {
+    super('cat', name, gender, 'meow!');
+    this.friends = ['Tom', 'Jenny', 'Selina'];
+  }
+}
+
+class CatWoman extends Human {
+  constructor(name, gender) {
+    super(name, gender);
+    this.friends = ['Tom', 'Felix'];
+    this.saying = new Cat().saying;
```

Comment:
Nice workaround. Works fine when a class we borrow a property from guarantees the property is set. This particular task implementation is the case.

**You do not need to fix anything in this regards**. Just the the below into consideration.

However, be careful. In general case you would want to borrow a class static property/method rather than a property of an individual `Cat`. Think of it as if you created a cat, took its saying and... well... discarded the cat itself. Doesn't look nice.

Yet another approach is to implement an 'independent' function or variable `meow` and use it to define a relevant property for cats and catwomen.

Yet another approach is to implement composition pattern. Extra materials in the task's repo refers to an article on this topic.

Reply:
About the "compositional" approach, I tried to understand it, but I discovered and started to explore JS OOP through a "Class" pattern and haven't fully covered it yet, so I want to fully understand it first and after other options (:

Reply:
@5Mountains ok. Then just leave this part as it is